packages/datadog-instrumentations/src/restify.js

```diff
@@ -50,7 +50,12 @@ function wrapFn (fn) {
     enterChannel.publish({ req, route })
 
     try {
-      return fn.apply(this, arguments)
+      const result = fn.apply(this, arguments)
+      if (result && typeof result === 'object' && typeof result.then === 'function') {
+        nextChannel.publish({ req })
+        finishChannel.publish({ req })
```

Comment:
Should it not be inside a `result.then(...)` so it will wait for it to actually resolve/reject?

Reply:
You're probably right

Reply:
@Qard I made the change, can you review? Not quite sure if I mucked up the arguments spread with my solution.